src/v2/view-builder/components/AuthenticatorOptions.js

```diff
@@ -0,0 +1,49 @@
+/*!
+ * Copyright (c) 2019, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+import { ListView, loc } from 'okta';
+import skipAll from './views/SkipOptionalEnrollmentButton';
+import FactorRow from './views/FactorRow';
+
+export default ListView.extend({
+
+  className: 'enroll-factor-list',
+
+  item: FactorRow,
+
+  itemSelector: '.list-content',
+
+  initialize: function () {
+    this.listenTo(this.collection,'selectFactor', function (data) {
+      this.model.set(this.options.name, data);
+      this.options.appState.trigger('saveForm', this.model);
+    });
+    this.hasOptionalFactors = this.options.appState.hasRemediationObject('skip');
+    if (this.hasOptionalFactors) {
+      this.model.set('formName', 'skip');
+      this.add(skipAll);
+    }
+  },
+
+  template: '<div class="list-content"> <div class="authenticator-list-title"> {{title}} </div> </div>',
+
+  getTemplateData () {
+    // presence of the skip remediation form tells us that the authenticators are all optional
+    let title = '';
+    if(this.options.appState.get('currentFormName') === 'select-authenticator-enroll') {
```

Comment:
To avoid checking on `currentFormName`, we can try to generate different component at uiSchemaTransformer.

Today we same `type` for verify and enroll: `type: 'authenticatorSelect',`. Imagine it could be
`type: authenticatorVerifySelect`
and `type: authenticatorEnrollSelect`

thereby creates `AuthenticatorVerifySelect.js` and `AuthenticatorEnrollSelect.js` component respectively.
In those two components, you won't need to check on `currentFormName` (as well as their child component) but just render view per mock. (You could share children components if they are Form agnostic)

Reply:
Thanks for that suggestion. It did occur to me that at some point we need to break the verify and enroll into different components.